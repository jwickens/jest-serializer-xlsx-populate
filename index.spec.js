const XlsxPopulate = require('xlsx-populate');

async function createTestData () {
  const wb = await XlsxPopulate.fromBlankAsync()
  wb.sheet("Sheet1").cell("A1").value("This is neat!")
  wb.addSheet("Sheet2")
  wb.sheet("Sheet2").cell("A1").value("Student")
  wb.sheet("Sheet2").cell("A2").value("John")
  wb.sheet("Sheet2").cell("A3").value("Sally")
  wb.sheet("Sheet2").cell("A4").value("Kim")
  wb.sheet("Sheet2").cell("B1").value("Grade")
  wb.sheet("Sheet2").cell("B2").value("F")
  wb.sheet("Sheet2").cell("B3").value("A+")
  wb.sheet("Sheet2").cell("B3").value(undefined)
  return wb
}

it('saves a snapshot', async () => {
  const d = await createTestData()
  expect(d).toMatchSnapshot()
})

it('ignores other data types', () => {
  expect({type: 'not a workbook'}).toMatchSnapshot()
})