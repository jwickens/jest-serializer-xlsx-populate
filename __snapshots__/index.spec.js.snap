// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ignores other data types 1`] = `
Object {
  "type": "not a workbook",
}
`;

exports[`saves a snapshot 1`] = `
Sheet1:

This is neat!
---
Sheet2:

Student  Grade
John     F
Sally
Kim
---

`;
